src/EnergyPlus/WaterThermalTanks.cc

```diff
@@ -7482,19 +7462,8 @@ namespace WaterThermalTanks {
 				// Reset the water heater's mode (call above may have changed modes)
 				Tank.Mode = HeatPump.SaveWHMode;
 
-				// Estimate portion of time step that the HP operates based on a linear interpolation of the tank temperature decay
-				// assuming that all heating sources are off.
-				if ( TankTemp != NewTankTemp ) {
-					HPPartLoadRatio = max( 0.0, min( 1.0, ( ( SetPointTemp - DeadBandTempDiff ) - NewTankTemp ) / ( TankTemp - NewTankTemp ) ) );
-				} else {
-					HPPartLoadRatio = 1.0;
-				}
+				HPPartLoadRatio = 1.0;
```

Comment:
This may be controversial, but I believe this to be correct. Here we're setting the PLR to see if it will overshoot the setpoint if it runs the whole timestep, which gets checked below. If we only have a 0.5 PLR estimated here, it could check that, see that it doesn't meet setpoint and assume it needs to be at a PLR of 1 later on when it really only needed to be at say 0.7. This causes the temperature to overshoot.